Formula/ipython.rb

```diff
@@ -188,51 +188,13 @@ class Ipython < Formula
   end
 
   def install
-    xy = Language::Python.major_minor_version Formula["python@3.10"].opt_bin/"python3"
-    ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python#{xy}/site-packages"
-
-    # install other resources
-    ipykernel = resource("ipykernel")
-    res = resources - [ipykernel]
-    res -= [resource("appnope")] if OS.linux?
-    res.each do |r|
-      r.stage do
-        system Formula["python@3.10"].opt_bin/"python3", *Language::Python.setup_install_args(libexec/"vendor")
-      end
-    end
-
-    # install and link IPython
-    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python#{xy}/site-packages"
-    system Formula["python@3.10"].opt_bin/"python3", *Language::Python.setup_install_args(libexec)
-    bin.install libexec/"bin/ipython"
-    bin.env_script_all_files(libexec/"bin", PYTHONPATH: ENV["PYTHONPATH"] + "${PYTHONPATH:+:}$PYTHONPATH")
-
-    # install IPython man page
-    man1.install libexec/"share/man/man1/ipython.1"
+    venv = virtualenv_create(libexec, "python3")
+    res = resources.reject { |r| r.name == "appnope" && OS.linux? }
+    venv.pip_install res
+    venv.pip_install_and_link buildpath
 
-    # install IPyKernel
-    ipykernel.stage do
-      system Formula["python@3.10"].opt_bin/"python3", *Language::Python.setup_install_args(libexec/"vendor")
-    end
-
-    # install kernel
-    kernel_dir = Dir.mktmpdir
-    system libexec/"bin/ipython", "kernel", "install", "--prefix", kernel_dir
-    (share/"jupyter/kernels/python3").install Dir["#{kernel_dir}/share/jupyter/kernels/python3/*"]
-    inreplace share/"jupyter/kernels/python3/kernel.json", "]", <<~EOS
```

Comment:
not sure if this is the right approach. but a simple venv install was replaced in https://github.com/Homebrew/homebrew-core/pull/14916#issuecomment-310930681. trying to follow the reasoning, it seems like it was to enable `jupyterlab` to depend on `ipython` but we don't do this anymore

Reply:
I think we do want to do that, though. It just wasn't finished off in #89374.